### PR TITLE
 Implement testing

### PR DESCRIPTION
**Pull request by virag-ky**
We implemented testing in the Math Magicians project.

- set up the React Testing Library
- wrote unit tests for operate.js and calculate.js
- use Jest snapshots to test the components

### CODE REVIEW

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @virag-ky and @aliatayee,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### In keeping with the required Gitflow from the guidelines, you are required to merge into the branch not the main branch.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/tests/Calculator.test.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    expect(calculator).toMatchSnapshot();
+  });
+
+  test('check title exist', () => {
+    render(<Calculator />);
+    expect(
+      screen.getByRole('heading', { name: /Let's do some math!/i }),
+    ).toBeInTheDocument();
+  });
+});
```
Comment (elisha2kyakpo1) [inline src/tests/Calculator.test.js:18]: - [ ] You have done well to test the Calculator code, but let's add more tests. All the functionalities & scenarios of the operate.js and calculate.js modules should be covered using Jest.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- The project PR title and description looks professional 👍
- No linter errors 👍
- Descriptive commit messages 👍

## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly add a command for testing the app in the readme file.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @virag-ky, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**